addons/l10n_latam_invoice_document/wizards/account_move_reversal.py

```diff
@@ -0,0 +1,65 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+
+
+class AccountMoveReversal(models.TransientModel):
+
+    _inherit = "account.move.reversal"
+
+    l10n_latam_use_documents = fields.Boolean(
+        related='move_id.journal_id.l10n_latam_use_documents', readonly=True)
+    l10n_latam_document_type_id = fields.Many2one('l10n_latam.document.type', 'Document Type', ondelete='cascade')
+    l10n_latam_sequence_id = fields.Many2one('ir.sequence', compute='_compute_l10n_latam_sequence')
+    l10n_latam_document_number = fields.Char(string='Document Number')
+
+    @staticmethod
+    def reverse_type_map(move_type):
+        match = {
+            'entry': 'entry',
+            'out_invoice': 'out_refund',
+            'in_invoice': 'in_refund',
+            'in_refund': 'in_invoice',
+            'out_receipt': 'in_receipt',
+            'in_receipt': 'out_receipt'}
+        return match.get(move_type)
+
+    @api.onchange('move_id')
+    def _onchange_move_id(self):
+        if self.move_id.l10n_latam_use_documents:
+            refund = self.move_id.new({
+                'type': self.reverse_type_map(self.move_id.type),
+                'journal_id': self.move_id.journal_id.id,
+                'partner_id': self.move_id.partner_id.id,
+                'company_id': self.move_id.company_id.id,
+            })
+            refund._compute_l10n_latam_documents()
+            self.l10n_latam_document_type_id = refund.l10n_latam_document_type_id
+            return {'domain': {
+                'l10n_latam_document_type_id': [('id', 'in', refund.l10n_latam_available_document_type_ids.ids)]}}
```

Comment:
o_0

all this to compute one domain? There must be easier solutions, don't you think?